sympy/solvers/solveset.py

```diff
@@ -1212,65 +1211,55 @@ def _solve_log(f, symbol):
     to exponents.
     """
 
-    lhs, rhs_s = invert_real(f, 0, symbol)
-    rhs = rhs_s.args[0]
-    new_f = None
-
+    lhs, rhs = f.lhs, f.rhs
     new_lhs = logcombine(lhs, force=True)
-    if new_lhs != lhs:
-        new_f = new_lhs - rhs
+    new_f = new_lhs - rhs
 
     return new_f
 
 
-def _is_logarithmic(f, symbol):
+def _is_logarithm_reducable_to_single_instance(f, symbol):
```

Comment:
Maybe `reducible`?

Reply:
My suggestion is to be verbose in the docstring where you define what it means to be logarithmic from the transolve point of view. It's not a matter of `expr.has(log)`, it is ... {insert definition}.